### ISSUE TITLE
GraphBLAS/Source/Generated/GB_AxB__any_first_bool.c:278:5: error: this 'if' clause does not guard... [-Werror=misleading-indentation]

### ISSUE THREAD

**Issue by marxin** [DrTimothyAldenDavis/SuiteSparse#84]
Using latest GCC 11, I can see the following error:

```
/home/abuild/rpmbuild/BUILD/SuiteSparse-5.9.0/GraphBLAS/Source/Template/GB_bitmap_AxB_saxpy_A_bitmap_B_sparse_template.c: In function 'GB_Asaxpy3B__any_first_bool':
/home/abuild/rpmbuild/BUILD/SuiteSparse-5.9.0/GraphBLAS/Source/Generated/GB_AxB__any_first_bool.c:278:5: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
  278 |     if (exists && !cb) cx = (ax) ; cb |= exists
      |     ^~
/home/abuild/rpmbuild/BUILD/SuiteSparse-5.9.0/GraphBLAS/Source/Template/GB_bitmap_AxB_saxpy_A_bitmap_B_sparse_template.c:327:33: note: in expansion of macro 'GB_BITMAP_MULTADD'
  327 |                                 GB_BITMAP_MULTADD (
      |                                 ^~~~~~~~~~~~~~~~~
/home/abuild/rpmbuild/BUILD/SuiteSparse-5.9.0/GraphBLAS/Source/Template/GB_bitmap_AxB_saxpy_A_bitmap_B_sparse_template.c:328:37: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
  328 |                                     Hf [pH+ii], Hx [pH+ii],
      |                                     ^~
/home/abuild/rpmbuild/BUILD/SuiteSparse-5.9.0/GraphBLAS/Source/Generated/GB_AxB__any_first_bool.c:278:36: note: in definition of macro 'GB_BITMAP_MULTADD'
  278 |     if (exists && !cb) cx = (ax) ; cb |= exists
      |                                    ^~
```

I see the project intentionally enables it with:
```
GraphBLAS/Demo/Include/graphblas_demos.h:#pragma GCC diagnostic error "-Wmisleading-indentation"
GraphBLAS/Source/GB_warnings.h:#pragma GCC diagnostic error "-Wmisleading-indentation"
```

**Comment by DrTimothyAldenDavis** [comment]
I don't have access to gcc 11 (the latest version that spack supports is gcc 10.2), so I can't test this right away.  However, I will add the curly brackets so the macro will be: 

` if (exists && !cb) { cx = (ax) ; } cb |= exists ;`

then I'll post and update and perhaps you could try it with gcc 11.

**Comment by marxin** [comment]
Sure, I can test it for you. Can you please prepare a patch? It seems the code is generated by `GraphBLAS/Source/codegen_axb_method.m`, but my tarball `SuiteSparse-5.9.0.tar.gz` contains quite many occurrences in the generated files.

**Comment by DrTimothyAldenDavis** [comment]
A patch would be difficult since there are many many files that changed.
I've posted an update to GraphBLAS here:
https://github.com/DrTimothyAldenDavis/GraphBLAS/tree/master

see for example the curly brackets here:
https://github.com/DrTimothyAldenDavis/GraphBLAS/blob/de8bae05798b5fa67f0dafdf810fdc562850bee0/Source/Generated/GB_AxB__any_first_fp32.c#L280

This version has all the files fixed with the addition of these curly
brackets.

**Comment by marxin** [comment]
> A patch would be difficult since there are many many files that changed.

I see. For now, I disabled the warning in my package and I'll enable it once a new release is released.
Thanks for the help.

**Comment by kilasuelika** [comment]
I have the same problem with you. I'm using the latest experimental GCC 12.

**Comment by DrTimothyAldenDavis** [comment]
This is fixed in GraphBLAS v5.0.2, but not yet included in the SuiteSparse
meta package, which still has v4.0.3.  This will be fixed once I include
v5.0.x in SuiteSparse in the next SuiteSparse release.  In the meantime,
you can just delete GraphBLAS v4.0.x from SuiteSparse and replace it with
v5.0.3 from the GraphBLAS-only github repo.

**Comment by DrTimothyAldenDavis** [comment]
I have included GraphBLAS v5.0.5 in SuiteSparse v5.10.1.  Can you give it a try with gcc 11 and the experimental 12?  The warnings should be silenced now but I can't check it myself.

**Comment by marxin** [comment]
> I have included GraphBLAS v5.0.5 in SuiteSparse v5.10.1. Can you give it a try with gcc 11 and the experimental 12? The warnings should be silenced now but I can't check it myself.

Sure, I'm going to test it.

**Comment by marxin** [comment]
I can confirm `v5.10.1` can be successfully built with GCC `11.1`. Thanks for it.
